pkg/helm/actions/upgrade_release.go

```diff
@@ -23,6 +23,12 @@ func UpgradeRelease(ns, name, url string, vals map[string]interface{}, conf *act
 		}
 		return nil, err
 	}
+
+	// Before proceeding, check if chart URL is present as an annotation
+	if chart_url, ok := rel.Chart.Metadata.Annotations["chart_url"]; url == "" && ok {
+		url = chart_url
+	}
+
 	// if url is not provided then we expect user trying to upgrade release with the same
 	// version of chart but with the different values
 	if url == "" {
```

Comment:
see above, can this happen now?

Reply:
I think it can if the release came from cli, for example